src/hardware/pcspeaker.cpp

```diff
@@ -442,31 +420,38 @@ void PCSPEAKER_SetType(const bool pit_clock_gate_enabled, const bool pit_output_
 }
 
 // TODO: check if this is accurate
-static inline float sinc(const float t)
+static double sinc(const double t)
 {
-#define SINC_ACCURACY 20
-	float result = 1;
+	constexpr auto SINC_ACCURACY = 20;
+	double result = 1;
 	for (auto k = 1; k < SINC_ACCURACY; ++k) {
-		result *= cosf(t / powf(2.0f, static_cast<float>(k)));
+		result *= cos(t / pow(2.0, k));
 	}
 	return result;
 }
 
-static inline float impulse(const float t)
+static float impulse(const double t)
 {
```

Comment:
Okay... I'm reviewing this commit by commit, now I can see you've switched back to doubles.

Reply:
Yup, for the impulse, which is used to populate the LUT.  
I was hoping I could make that function constexpr too.. (so the LUT is computed at compile-time), but the trig functions aren't constexpr on all compilers.  I think c++20 finally fixes that.